File: Java/66.java
__________________________________________________________________________________________________
sample 0 ms submission
class Solution {
    public int[] plusOne(int[] digits) {        
    int n = digits.length;
    for(int i=n-1; i>=0; i--) {
        if(digits[i] < 9) {
            digits[i]++;
            return digits;
        }
        
        digits[i] = 0;
    }
    
    int[] newNumber = new int [n+1];
    newNumber[0] = 1;
    
    return newNumber;
}
}
__________________________________________________________________________________________________
sample 34596 kb submission
class Solution {
    public int[] plusOne(int[] digits) {
		  HashMap<Integer, Integer> mp=new HashMap<Integer,Integer>();
		  int j=0,i=0,carry=1,temp=0;
		  for ( j=digits.length-1;j>=0;j--)
		  { 
			    	  temp=((digits[j])+carry)%10;
			    	  if((digits[j])+carry==10)
			    	  carry=1;
			    	  else carry=0;
			      
			    	  mp.put(i,temp);
			    	  i++;
		  }
		  if (carry==1)
			  mp.put(i,carry);

		  int[] newarr=new int[mp.size()];
		  i=0;
		  for (int p=mp.size()-1;p>=0;p--)
		  {
			  newarr[i]=mp.get(p);
			  i++;	
		  }  
		  
		  return newarr;    
	    
    }
}
__________________________________________________________________________________________________

File: Python3/66.py
__________________________________________________________________________________________________
sample 16 ms submission
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        if digits[-1] != 9:
            digits[-1] += 1
        else:
            found = False
            for i in range(len(digits)-1,-1,-1):
                if digits[i] == 9:
                    digits[i] = 0
                else:
                    digits[i] += 1
                    found = True
                    break
            if not found:
                digits = [1] + digits
        
        return digits
__________________________________________________________________________________________________
sample 12908 kb submission
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        
        s = ""
        
        for d in digits:
            s += str(d)
        
        num = int(str(s)) + 1 
        
        return [int(d) for d in str(num)]            
__________________________________________________________________________________________________
